visualization/app/codeCharta/util/color/colorConverter.ts

```diff
@@ -3,6 +3,25 @@ import convert from "color-convert"
 import { HSL } from "./hsl"
 
 export class ColorConverter {
+	private static colorToVector3Map = new Map<string, Vector3>()
+	private static hexToNumberMap = new Map<string, number>()
+
+	public static getVector3(color: string): Vector3 {
+		if (!this.colorToVector3Map.has(color)) {
+			this.colorToVector3Map.set(color, ColorConverter.colorToVector3(color))
+		}
+
+		return this.colorToVector3Map.get(color)
```

Comment:
```suggestion
		let vector = this.colorToVector3Map.get(color)
		if (vector === undefined) {
			vector = ColorConverter.colorToVector3(color)
			this.colorToVector3Map.set(color, vector)
		}

		return vector
```

This reduces the lookups from two to one.